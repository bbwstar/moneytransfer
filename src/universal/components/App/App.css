body{
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  font-family: 'Roboto Condensed', sans-serif;
  font-family: 'Roboto Slab', serif;
  font-family: 'Roboto Mono', monospace;
  font-family: 'Roboto', sans-serif;
}
