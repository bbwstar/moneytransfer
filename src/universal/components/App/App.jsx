import React from 'react';
import PropTypes from 'prop-types';

import Navigation from 'components/Navigation/Navigation';
import 'styles/global.css';
import styles from './App.css';

const App = props =>
  (<div className={styles.app}>

    {props.location.pathname === '' ? <Navigation /> : ''}
    {props.children}

  </div>);

App.propTypes = {
  children: PropTypes.element.isRequired,
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired,
};

export default App;
