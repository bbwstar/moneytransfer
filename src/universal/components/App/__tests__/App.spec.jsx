import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

jest.mock('components/Navigation/Navigation', () => 'Navigation');

describe('App (Snapshot)', () => {
  it('renders without navigation', () => {
    const component = renderer.create(
      <App location={{ pathname: '/review' }}>
        <div />
      </App>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with navigation', () => {
    const component = renderer.create(
      <App location={{ pathname: '' }}>
        <div />
      </App>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});

