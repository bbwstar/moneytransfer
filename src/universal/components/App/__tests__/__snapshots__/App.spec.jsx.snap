// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App (Snapshot) renders with navigation 1`] = `
<div
  className="app"
>
  <Navigation />
  <div />
</div>
`;

exports[`App (Snapshot) renders without navigation 1`] = `
<div
  className="app"
>
  
  <div />
</div>
`;
